### PR TITLE
Move build things around and add IPDK eBPF container and native build

### PR DESCRIPTION
**Pull request by mestery**
This commit does 3 things really:

1. Moves some of the build infrastructure out of the networking
   directory and into the build directory.
2. Fixes a bunch of shellcheck errors.
3. Add networking_ebpf directory and recipe code.

The networking_ebpf code for now is about a few things:

1. Clones, builds, and installs the protobuf repository.
2. Clones, builds, and installs the p4c compiler with ebpf backend
   support.
3. Clones, builds, and installs the psabpf CLI tool.
4. Clones the psa-ebpf-demo repository.
5. Installs golang.
6. Clones, builds, and installs the ipdk-tap-plugin Docker CNM.

The networking_ebpf container is a bit of a work in progress now. I'd also like feedback on if I should break out the build changes from the networking_ebpf recipe as well.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

**Comment by mestery** [comment]
@stolsma Let me know your thoughts on this one in terms of splitting out the build changes or not. They are sort of required to add the `networking_ebpf` container build, so I have them together here.

**Comment by mestery** [comment]
Very nice, the new `ebpf-ubuntu2004` CI job ran successfully! Kudos to @stolsma for the awesome work on the `ipdk` CLI and build. You know it was designed well when adding in a new target was as simple as a few lines of changes to the build code. :)

**Comment by stolsma** [comment]
@mestery that was the whole idea when creating the cli 🤷‍♂️😊

Will look at this PR in detail tomorrow morning CET. but first impression is 👍

File: build/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# IPDK Build Directory
+
+This directory contains all of the individual IPDK recipes. These can all be
+built using the `ipdk` CLI found here.
+
+## Install IPDK CLI
+To fully use all the features of IPDK container and if you want to follow the
```
Comment (stolsma) [inline build/README.md:7]:
```suggestion
To fully use all the features of the IPDK containers and if you want to follow the
```

**Review by stolsma — COMMENTED** [review]
Only one language comment (just minor, throw away when not applicable). 

Didn't try the different containers to see if they still run, will do that next week. Have some other stuff to do today but I think this should be good enough to LGTM....

**Review by stolsma — APPROVED** [review]

File: build/networking_ebpf/README_DOCKER.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ built with the following components:
 3. [p4c eBPF PSA implementation](https://github.com/p4lang/p4c/tree/main/backends/ebpf/psa)
 4. [psabpf CLI](https://github.com/P4-Research/psabpf)
 5. [PSA eBPF Demo](https://github.com/P4-Research/psa-ebpf-demo)
-6. [IPDK Docker CNM](https://github.com/mestery/ipdk-tap-plugin) (Work in progress)
+6. [IPDK Docker CNM](https://github.com/mestery/ipdk-plugin) (Work in progress)
```
Comment (stolsma) [inline build/networking_ebpf/README_DOCKER.md:12]: Should CNM not be CNI (Container Network Interface)?

Comment (mestery) [reply build/networking_ebpf/README_DOCKER.md:12]: Good catch, fixed.

**Review by stolsma — APPROVED** [review]
I did a quick check and for the first commit of eBPF vSwitch networking support it looks great! So for me a LGTM! (caveat: I didn't test it on my computer so I don't know if everything really works as planned)

**Review by dandaly — APPROVED** [review]
This looks great!

**Pull request merged**